### PR TITLE
Accept softfails as passed and calculate result correctly

### PR DESCRIPTION
**Pull request by aaannz**

### CODE REVIEW

File: t/10-jobs.t [hunk 1]
```diff
@@ -43,4 +43,147 @@ ok(grep(!/^(99962|99945)$/, @ids));
...
+    $job->update;
+    $job->discard_changes;
+
+    is($job->result, OpenQA::Schema::Result::Jobs::NONE, 'result is not yet set');
+    $job->done;
+    $job->discard_changes;
+    is($job->result, OpenQA::Schema::Result::Jobs::FAILED, 'job result is failed');
+};
+
+subtest 'job with all important modules passed and at least one unimportant failed =? overall passed' => sub {
```
Comment (okurz) [inline t/10-jobs.t]: the arrow in this line turned into a =?

File: t/10-jobs.t [hunk 2]
```diff
@@ -43,4 +43,147 @@ ok(grep(!/^(99962|99945)$/, @ids));
...
+    $job->update_module('d', {result => 'fail', details => []});
+    $job->update;
+    $job->discard_changes;
+    is($job->result, OpenQA::Schema::Result::Jobs::NONE, 'result is not yet set');
+    $job->done;
+    $job->discard_changes;
+    is($job->result, OpenQA::Schema::Result::Jobs::PASSED, 'job result is passed');
+};
+
+subtest 'job with important modules passed and at least one softfailed and at least one unimportant failed =? overall softfailed' => sub {
```
Comment (okurz) [inline t/10-jobs.t]: Here, too

File: t/10-jobs.t [hunk 3]
```diff
@@ -43,4 +43,147 @@ ok(grep(!/^(99962|99945)$/, @ids));
...
+    $job->update_module('d', {result => 'fail', details => []});
+    $job->update;
+    $job->discard_changes;
+    is($job->result, OpenQA::Schema::Result::Jobs::NONE, 'result is not yet set');
+    $job->done;
+    $job->discard_changes;
+    is($job->result, OpenQA::Schema::Result::Jobs::SOFTFAILED, 'job result is softfailed');
+};
+
+subtest 'job with one important module failed and at least one unimportant passed =? overall failed' => sub {
```
Comment (okurz) [inline t/10-jobs.t]: And here

**Comment by okurz** [comment]
Very nice to see these tests, a bit copy-pasty but nice :-)

**Comment by aaannz** [comment]
rebased and tests fixed

**Comment by aaannz** [comment]
Interesting that that topic line wasn't copy pasted at all. I just forgot my glasses at home and eventually got tired of reading what I'm writing. ;)

**Comment by okurz** [comment]
I guess the tests could still use more helper functions to reduce duplicate code and reduce readability but LGTM anyway

**Pull request merged**